sdk/core/Azure.Core/src/Shared/RequestContentHelper.cs

```diff
@@ -48,8 +58,13 @@ public static RequestContent FromEnumerable(IEnumerable<BinaryData> enumerable)
             return content;
         }
 
-        public static RequestContent FromDictionary<T>(IDictionary<string, T> dictionary) where T : notnull
+        public static RequestContent? FromDictionary<T>(IDictionary<string, T> dictionary) where T : notnull
```

Comment:
If it turns out we do want these changes, won't we need to update the nullability of the parameter?

Reply:
But seems the `where T : notnull` is also useless since we actually call this method with a dictionary of nullable value type. Maybe we could remove this as well.